extensions/cornerstone/src/Viewport/Overlays/CustomizableViewportOverlay.tsx

```diff
@@ -0,0 +1,532 @@
+  };
+
+  // calculated values
+  voi: {
+    windowWidth: number;
+    windowCenter: number;
+  };
+  instanceNumber?: number;
+  scale?: number;
+}
+
+/**
+ * Window Level / Center Overlay item
+ */
+function VOIOverlayItem({ voi, customization }: OverlayItemProps) {
+  const { windowWidth, windowCenter } = voi;
+  if (typeof windowCenter !== 'number' || typeof windowWidth !== 'number') {
+    return null;
+  }
+
+  return (
+    <div
+      className="overlay-item flex flex-row"
+      style={{ color: (customization && customization.color) || undefined }}
+    >
+      <span className="mr-1 shrink-0">W:</span>
+      <span className="ml-1 mr-2 font-light shrink-0">
+        {windowWidth.toFixed(0)}
+      </span>
+      <span className="mr-1 shrink-0">L:</span>
+      <span className="ml-1 font-light shrink-0">
+        {windowCenter.toFixed(0)}
+      </span>
+    </div>
+  );
+}
+
+/**
+ * Zoom Level Overlay item
+ */
+function ZoomOverlayItem({ scale, customization }: OverlayItemProps) {
+  return (
+    <div
+      className="overlay-item flex flex-row"
+      style={{ color: (customization && customization.color) || undefined }}
+    >
+      <span className="mr-1 shrink-0">Zoom:</span>
+      <span className="font-light">{scale.toFixed(2)}x</span>
+    </div>
+  );
+}
+
+/**
+ * Instance Number Overlay Item
+ */
+function InstanceNumberOverlayItem({
+  instanceNumber,
+  imageSliceData,
+  customization,
+}: OverlayItemProps) {
+  const { imageIndex, numberOfSlices } = imageSliceData;
+
+  return (
+    <div
+      className="overlay-item flex flex-row"
+      style={{ color: (customization && customization.color) || undefined }}
+    >
+      <span className="mr-1 shrink-0">I:</span>
+      <span className="font-light">
+        {instanceNumber !== undefined && instanceNumber !== null
+          ? `${instanceNumber} (${imageIndex + 1}/${numberOfSlices})`
+          : `${imageIndex + 1}/${numberOfSlices}`}
+      </span>
+    </div>
+  );
+}
+
+/**
+ * Customizable Viewport Overlay
+ */
+function CustomizableViewportOverlay({
+  element,
+  viewportData,
+  imageSliceData,
+  viewportIndex,
+  servicesManager,
+}) {
+  const {
+    ToolBarService,
+    CornerstoneViewportService,
+    customizationService,
+  } = servicesManager.services;
+  const [voi, setVOI] = useState({ windowCenter: null, windowWidth: null });
+  const [scale, setScale] = useState(1);
+  const [activeTools, setActiveTools] = useState([]);
+  const { imageIndex } = imageSliceData;
+
+  const topLeftCustomization = customizationService.getModeCustomization(
+    'cornerstoneOverlayTopLeft'
+  );
+  const topRightCustomization = customizationService.getModeCustomization(
+    'cornerstoneOverlayTopRight'
+  );
+  const bottomLeftCustomization = customizationService.getModeCustomization(
+    'cornerstoneOverlayBottomLeft'
+  );
+  const bottomRightCustomization = customizationService.getModeCustomization(
+    'cornerstoneOverlayBottomRight'
+  );
```

Comment:
I didn't expect to see these `getModeCus...`, why we are not just saying `getCustomization`?

Reply:
At the moment of writing, the `CustomizationService` only defines `getModeCustomization()` function